realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -708,17 +708,35 @@ bool BaseRealSenseNode::setBaseTime(double frame_time, rs2_timestamp_domain time
     return false;
 }
 
+uint64_t BaseRealSenseNode::millisecondsToNanoseconds(double timestamp_ms)
+{
+        // modf breaks input into an integral and fractional part
+        double int_part_ms, fract_part_ms;
+        fract_part_ms = modf(timestamp_ms, &int_part_ms);
+
+        //convert to ns
+        uint64_t int_part_ns = static_cast<uint64_t>(int_part_ms) * 1000000;
+        uint64_t fract_part_ns = static_cast<uint64_t>(fract_part_ms * 10000000);
+
+        // Convert fract_parts_ns into 6 digits and round it
+        // to be aligned with librealsense get_timestamp API
+        fract_part_ns = (fract_part_ns % 10 > 4) ? (fract_part_ns/10 + 1) : (fract_part_ns/10);
```

Comment:
More efficient `( fract_part_ns + 5 ) / 10`. No need to use `? :`.
Or use `round` function while converting to `uint64_t`